lib/cerner/oauth1a/access_token.rb

```diff
@@ -1,70 +1,138 @@
 # frozen_string_literal: true
 
+require 'cerner/oauth1a/oauth_error'
+require 'cerner/oauth1a/protocol'
+require 'uri'
+
 module Cerner
   module OAuth1a
 
-    # Public: An OAuth 1.0a Access Token.
+    # Public: A Cerner OAuth 1.0a Access Token and related request parameters for use in Consumer or
+    # Service Provider use cases.
     class AccessToken
+      # Public: Constructs an AccessToken using the value of an HTTP Authorization Header based on
+      # the OAuth HTTP Authorization Scheme (https://oauth.net/core/1.0a/#auth_header).
+      #
+      # value - A String containing the HTTP Authorization Header value.
+      #
+      # Returns an AccessToken.
+      #
+      # Raises a Cerner::OAuth1a::OAuthError with a populated oauth_problem if any of the parameters
+      # in the value are invalid.
+      def self.from_authorization_header(value)
+        params = Protocol.parse_authorization_header(value)
+
+        raise OAuthError.new('', nil, 'version_rejected') if params[:oauth_version] && params[:oauth_version] != '1.0'
```

Comment:
Any reason to not do:
`raise OAuthError.new('', nil, 'version_rejected') unless params[:oauth_version] = '1.0'`

Reply:
@johnlcox, small change, it would be `unless params[:oauth_version] == '1.0'` or `unless params[:oauth_version].eql? '1.0'`

Reply:
Missing the double equals like I did in my example might be a good reason to avoid doing that. I wasn't aware of the `eql?` function.

Reply:
'oauth_version' is an optional parameter and when it's missing, the code is supposed to assume it's 1.0.